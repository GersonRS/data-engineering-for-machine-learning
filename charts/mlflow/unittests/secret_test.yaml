suite: test secret

templates:
  - secret.yaml

tests:
  - it: should have user and password when postgres is enabled
    set:
      backendStore.postgres.enabled: true
      backendStore.postgres.user: postgres
      backendStore.postgres.password: P455w0rd!
    asserts:
      - equal:
          path: data.PGUSER
          value: cG9zdGdyZXM=
      - equal:
          path: data.PGPASSWORD
          value: UDQ1NXcwcmQh

  - it: should have user and password when mysql is enabled
    set:
      backendStore.mysql.enabled: true
      backendStore.mysql.user: mysql
      backendStore.mysql.password: P455w0rd!
    asserts:
      - equal:
          path: data.MYSQL_USERNAME
          value: bXlzcWw=
      - equal:
          path: data.MYSQL_PWD
          value: UDQ1NXcwcmQh

  - it: should have azure connection string when azureBlob is enabled and connection string set
    set:
      artifactRoot.azureBlob.enabled: true
      artifactRoot.azureBlob.container: mlflow
      artifactRoot.azureBlob.storageAccount: devstoreaccount1
      artifactRoot.azureBlob.connectionString: DefaultEndpointsProtocol=http;AccountName=devstoreaccount1;AccountKey=Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==;BlobEndpoint=http://azurite:10000/devstoreaccount1;QueueEndpoint=http://azurite:10001/devstoreaccount1
    asserts:
      - equal:
          path: data.AZURE_STORAGE_CONNECTION_STRING
          value: RGVmYXVsdEVuZHBvaW50c1Byb3RvY29sPWh0dHA7QWNjb3VudE5hbWU9ZGV2c3RvcmVhY2NvdW50MTtBY2NvdW50S2V5PUVieTh2ZE0wMnhOT2NxRmxxVXdKUExsbUV0bENEWEoxT1V6RlQ1MHVTUlo2SUZzdUZxMlVWRXJDejRJNnRxL0sxU1pGUFRPdHIvS0JIQmVrc29HTUd3PT07QmxvYkVuZHBvaW50PWh0dHA6Ly9henVyaXRlOjEwMDAwL2RldnN0b3JlYWNjb3VudDE7UXVldWVFbmRwb2ludD1odHRwOi8vYXp1cml0ZToxMDAwMS9kZXZzdG9yZWFjY291bnQx

  - it: should have azure storage access key when azureBlob is enabled and access key set
    set:
      artifactRoot.azureBlob.enabled: true
      artifactRoot.azureBlob.container: mlflow
      artifactRoot.azureBlob.storageAccount: devstoreaccount1
      artifactRoot.azureBlob.accessKey: Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==
    asserts:
      - equal:
          path: data.AZURE_STORAGE_ACCESS_KEY
          value: RWJ5OHZkTTAyeE5PY3FGbHFVd0pQTGxtRXRsQ0RYSjFPVXpGVDUwdVNSWjZJRnN1RnEyVVZFckN6NEk2dHEvSzFTWkZQVE90ci9LQkhCZWtzb0dNR3c9PQ==

  - it: should have aws access key id and secret access key when s3 is enabled and parameters set
    set:
      artifactRoot.s3.enabled: true
      artifactRoot.s3.bucket: test-bucket
      artifactRoot.s3.path: mlflow_folder
      artifactRoot.s3.awsAccessKeyId: AKIAIOSFODNN7EXAMPLE
      artifactRoot.s3.awsSecretAccessKey: wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY
    asserts:
      - equal:
          path: data.AWS_ACCESS_KEY_ID
          value: QUtJQUlPU0ZPRE5ON0VYQU1QTEU=
      - equal:
          path: data.AWS_SECRET_ACCESS_KEY
          value: d0phbHJYVXRuRkVNSS9LN01ERU5HL2JQeFJmaUNZRVhBTVBMRUtFWQ==

  - it: should match snapshot of default values
    release:
      name: mlflow
      namespace: mlflow
    chart:
      version: 1.0.0
      appVersion: 1.0.0
    asserts:
      - matchSnapshot: { }
