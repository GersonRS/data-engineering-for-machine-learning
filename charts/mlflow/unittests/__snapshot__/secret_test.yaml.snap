should match snapshot of default values:
  1: |
    apiVersion: v1
    data: null
    kind: Secret
    metadata:
      labels:
        app: mlflow
        chart: mlflow-1.0.0
        heritage: Helm
        release: mlflow
      name: mlflow-env-secret
    type: Opaque
